# Quarterly Sales Tax Revenue Analysis

### [Data Source](https://tax.utah.gov/econstats/sales)

### Q1: January, February, March

In [1]:
import pandas as pd
import os
import numpy as np
import arcpy
from arcgis.features import GeoAccessor, GeoSeriesAccessor
arcpy.env.overwriteOutput = True

In [2]:
# show all columns
pd.options.display.max_columns = None

In [3]:
# Removes misc characters from string
def replace(string):
    string = string.replace("&", "")
    string = string.replace("-", " ")
    string = string.replace("(", "")
    string = string.replace(")", "")
    string = string.replace("/", " ")
    string = string.replace(",", "")
    string = string.replace(".", "")
    string = string.replace(" ", "_")
    return string


### Load Q2 2020 sales tax revenue data

In [4]:
# load excel sheet into a dataframe
xlsx = '.\\Inputs\\2020-01-quarterly-sales.xlsx'
q1_2020 = pd.read_excel(xlsx, sheet_name='Table 9', header=5)

# Drop last empty two rows
q1_2020.drop(q1_2020.tail(2).index,inplace=True)

# forward fill values from merged cells
q1_2020['County'] = q1_2020['County'].fillna(method='ffill')
q1_2020['Location Code'] = q1_2020['Location Code'].fillna(method='ffill')
q1_2020['City'] = q1_2020['City'].fillna(method='ffill')

# Figures with less than 10 taxpayers have been rounded up per Tax Commission disclosure rules.
# Rename column and convert values to boolean
q1_2020.rename(columns={'Unnamed: 5':'Rounded Up'}, inplace=True)
crosswalk = {'*':True, np.nan:False} # County_ID
q1_2020['Rounded Up'] = q1_2020['Rounded Up'].map(crosswalk)

q1_2020.head()


County  Location Code         City  \
0  Beaver County         1002.0  Beaver City   
1  Beaver County         1002.0  Beaver City   
2  Beaver County         1002.0  Beaver City   
3  Beaver County         1002.0  Beaver City   
4  Beaver County         1002.0  Beaver City   

                     Economic Sector (NAICS Code)     2020Q1  Rounded Up  
0   AGRICULTURE, FORESTRY, FISHING & HUNTING (11)    30000.0        True  
1  MINING, QUARRYING, & OIL & GAS EXTRACTION (21)     3000.0        True  
2                                  UTILITIES (22)   900000.0        True  
3                               CONSTRUCTION (23)    20000.0        True  
4                           MANUFACTURING (31-33)  1158075.0       False

In [5]:
# pivot the table to get sectors as columns with tax values by city
sectors_q1_2020 = pd.pivot_table(q1_2020,values = '2020Q1',index ='City', columns = 'Economic Sector (NAICS Code)', aggfunc='first')

# fill NAs with 0
sectors_q1_2020 = sectors_q1_2020.fillna(0)


# fix column names
#new_names = [replace(item) for item in list(sectors_q1_2020.columns)]
new_names = ['L_ACCOMMODATION_20', 'S_ADMIN_SUPPORT_20','A_AG_WILDLIFE_20','L_CULTURAL_REC_20','C_CONSTRUCTION_20',
'E_EDUCATION_20','S_FINANCIAL_20','L_RSTRNT_BAR_20','H_HEALTH_CARE_20','S_IT_DATA_20','S_CORPORATE_MGMT_20',
'M_MANUFACTURING_20','E_EXTRACTION_20','O_OTHER_20','X_ADJUSTMENTS_20','R_AUTO_PRIVATE_20','S_PROF_TECH_SRV_20',
'G_GOVERNMENT_20','S_RENTAL_LEASING_20','R_BUILDING_SUPPLY_20','R_CLOTHING_20','R_ELECTRONICS_20','R_GROCERY_BEV_20',
'R_FURNITURE_20','R_GAS_STATIONS_20','R_GENERAL_RETAIL_20','R_HEALTH_RETAIL_20','R_OTHER_RETAIL_20','R_AUTO_RETAIL_20',
'R_NONSTORE_RETAIL_20','R_SPORT_HOBBY_20','L_SPECIAL_EVENT_20','W_DISTRIBUTION_20','X_UNKNOWN_20','U_UTILITIES_20',
'W_WHLSALE_DURABLE_20','S_WHLSLE_ETRADE_20','W_WHLSLE_NDURABLE_20']
sectors_q1_2020.columns = new_names

# get the total sales tax revenue
sectors_q1_2020['TOTAL_20'] = sectors_q1_2020[new_names].sum(axis=1)

sectors_q1_2020.head()

L_ACCOMMODATION_20  S_ADMIN_SUPPORT_20  A_AG_WILDLIFE_20  \
City                                                                      
American Fork            461346.0            583528.0          200000.0   
Beaver City              864111.0             22157.0           30000.0   
Blanding                 350343.0             34733.0            1000.0   
Bountiful                200000.0           2097686.0           15000.0   
Brigham                  858416.0            184456.0            3000.0   

               L_CULTURAL_REC_20  C_CONSTRUCTION_20  E_EDUCATION_20  \
City                                                                  
American Fork          1672492.0          8549541.0         77529.0   
Beaver City               8000.0            20000.0         20000.0   
Blanding                 25000.0           250000.0        150000.0   
Bountiful              1002730.0          1864898.0        139922.0   
Brigham                 350000.0           726011.0         52422.0   

               S_FINANCIAL_20  L_RSTRNT_BAR_20  H_HEALTH_CARE_20  \
City                                                               
American Fork       3192392.0       23701728.0          470936.0   
Beaver City          254817.0        2013748.0           60000.0   
Blanding              22597.0         400000.0           60000.0   
Bountiful           1191614.0       12291550.0         1168394.0   
Brigham              404093.0        5940942.0          346590.0   

               S_IT_DATA_20  S_CORPORATE_MGMT_20  M_MANUFACTURING_20  \
City                                                                   
American Fork     8512406.0               9000.0           3775746.0   
Beaver City        435101.0               6000.0           1158075.0   
Blanding           441230.0               1000.0            240368.0   
Bountiful         6436583.0              30000.0           2642459.0   
Brigham           2755109.0               8000.0           5284338.0   

               E_EXTRACTION_20  O_OTHER_20  X_ADJUSTMENTS_20  \
City                                                           
American Fork          80000.0   9961976.0         1819138.0   
Beaver City             3000.0    788673.0          264908.0   
Blanding                   0.0    269921.0         -625850.0   
Bountiful              15000.0   6455142.0         1649557.0   
Brigham                 1000.0   3297982.0          585715.0   

               R_AUTO_PRIVATE_20  S_PROF_TECH_SRV_20  G_GOVERNMENT_20  \
City                                                                    
American Fork          2000000.0           4266418.0          15000.0   
Beaver City             350000.0            116728.0              0.0   
Blanding                600000.0            103492.0         700000.0   
Bountiful              3000000.0           3263809.0          70000.0   
Brigham                2000000.0            694141.0           1000.0   

               S_RENTAL_LEASING_20  R_BUILDING_SUPPLY_20  R_CLOTHING_20  \
City                                                                      
American Fork            3427382.0            17263059.0      5007232.0   
Beaver City                80776.0              497496.0        45083.0   
Blanding                   60548.0             1307977.0        51653.0   
Bountiful                1900181.0             1111674.0      2381625.0   
Brigham                  1562099.0             1522381.0       368586.0   

               R_ELECTRONICS_20  R_GROCERY_BEV_20  R_FURNITURE_20  \
City                                                                
American Fork         6482038.0         4627911.0       3282013.0   
Beaver City             29368.0         1750000.0          4000.0   
Blanding               115116.0         2000000.0          3000.0   
Bountiful             3224014.0        22666281.0       1464503.0   
Brigham               1316243.0        11284680.0        531658.0   

               R_GAS_STATIONS_20  R_GE

### Load Q2 2019 sales tax revenue data

In [6]:
# load excel sheet into a dataframe
xlsx = '.\\Inputs\\2019-01-quarterly-sales.xlsx'
q1_2019 = pd.read_excel(xlsx, sheet_name='Table 9', header=5)

# Drop last empty two rows
q1_2019.drop(q1_2019.tail(2).index,inplace=True)

# forward fill values from merged cells
q1_2019['County'] = q1_2019['County'].fillna(method='ffill')
q1_2019['Location Code'] = q1_2019['Location Code'].fillna(method='ffill')
q1_2019['City'] = q1_2019['City'].fillna(method='ffill')

# Figures with less than 10 taxpayers have been rounded up per Tax Commission disclosure rules.
# Rename column and convert values to boolean
q1_2019.rename(columns={'Unnamed: 5':'Rounded Up'}, inplace=True)
crosswalk = {'*':True, np.nan:False} # County_ID
q1_2019['Rounded Up'] = q1_2019['Rounded Up'].map(crosswalk)

q1_2019.tail()

County  Location Code         City  \
2393  Weber County        29040.0  South Ogden   
2394  Weber County        29040.0  South Ogden   
2395  Weber County        29040.0  South Ogden   
2396  Weber County        29040.0  South Ogden   
2397  Weber County        29040.0  South Ogden   

                          Economic Sector (NAICS Code)     2019Q1  Rounded Up  
2393                               ACCOMMODATION (721)    30000.0        True  
2394             FOOD SERVICES & DRINKING PLACES (722)  9660230.0       False  
2395  OTHER SERVICES-EXCEPT PUBLIC ADMINISTRATION (81)  1380064.0       False  
2396                       PRIVATE MOTOR VEHICLE SALES  1250000.0        True  
2397                   PRIOR-PERIOD PAYMENTS & REFUNDS   770844.0       False

In [7]:
# pivot the table to get sectors as columns with tax values by city
sectors_q1_2019 = pd.pivot_table(q1_2019,values = '2019Q1',index ='City', columns = 'Economic Sector (NAICS Code)', aggfunc='first')

# fill NAs with 0
sectors_q1_2019 = sectors_q1_2019.fillna(0)

# fix column names
#new_names = [replace(item) for item in list(sectors_q1_2019.columns)]
new_names = ['L_ACCOMMODATION_19', 'S_ADMIN_SUPPORT_19','A_AG_WILDLIFE_19','L_CULTURAL_REC_19','C_CONSTRUCTION_19',
'E_EDUCATION_19','S_FINANCIAL_19','L_RSTRNT_BAR_19','H_HEALTH_CARE_19','S_IT_DATA_19','S_CORPORATE_MGMT_19',
'M_MANUFACTURING_19','E_EXTRACTION_19','O_OTHER_19','X_ADJUSTMENTS_19','R_AUTO_PRIVATE_19','S_PROF_TECH_SRV_19',
'G_GOVERNMENT_19','S_RENTAL_LEASING_19','R_BUILDING_SUPPLY_19','R_CLOTHING_19','R_ELECTRONICS_19','R_GROCERY_BEV_19',
'R_FURNITURE_19','R_GAS_STATIONS_19','R_GENERAL_RETAIL_19','R_HEALTH_RETAIL_19','R_OTHER_RETAIL_19','R_AUTO_RETAIL_19',
'R_NONSTORE_RETAIL_19','R_SPORT_HOBBY_19','L_SPECIAL_EVENT_19','W_DISTRIBUTION_19','X_UNKNOWN_19','U_UTILITIES_19',
'W_WHLSALE_DURABLE_19','S_WHLSLE_ETRADE_19','W_WHLSLE_NDURABLE_19']
sectors_q1_2019.columns = new_names

# get the total sales tax revenue
sectors_q1_2019['TOTAL_19'] = sectors_q1_2019[new_names].sum(axis=1)

sectors_q1_2019.head()

L_ACCOMMODATION_19  S_ADMIN_SUPPORT_19  A_AG_WILDLIFE_19  \
City                                                                      
American Fork            600000.0            619006.0          150000.0   
Beaver City              838216.0             23482.0           15000.0   
Blanding                 479889.0             33014.0            1000.0   
Bountiful                200000.0           2472059.0            1000.0   
Brigham                 1250000.0            179287.0            1000.0   

               L_CULTURAL_REC_19  C_CONSTRUCTION_19  E_EDUCATION_19  \
City                                                                  
American Fork          1655844.0          6481514.0         89078.0   
Beaver City               7000.0            80000.0         20000.0   
Blanding                 25000.0           124403.0         35000.0   
Bountiful              1162941.0          1810519.0        115181.0   
Brigham                 350000.0          2312924.0         52439.0   

               S_FINANCIAL_19  L_RSTRNT_BAR_19  H_HEALTH_CARE_19  \
City                                                               
American Fork       2616295.0       23486666.0          478232.0   
Beaver City          198107.0        2071292.0           45000.0   
Blanding              21600.0         400000.0           50000.0   
Bountiful           1163654.0       12483316.0          990965.0   
Brigham              194102.0        6096734.0          348583.0   

               S_IT_DATA_19  S_CORPORATE_MGMT_19  M_MANUFACTURING_19  \
City                                                                   
American Fork     9027053.0              15000.0           1629561.0   
Beaver City        697839.0               2000.0            683402.0   
Blanding           835190.0               1000.0            128951.0   
Bountiful         5098392.0              30000.0           2888286.0   
Brigham           2900344.0              20000.0           4667069.0   

               E_EXTRACTION_19  O_OTHER_19  X_ADJUSTMENTS_19  \
City                                                           
American Fork          90000.0  10323341.0         2235225.0   
Beaver City                0.0    775040.0          242275.0   
Blanding                   0.0    257825.0          162960.0   
Bountiful              15000.0   6313370.0         2545449.0   
Brigham                 1000.0   3207128.0         3453277.0   

               R_AUTO_PRIVATE_19  S_PROF_TECH_SRV_19  G_GOVERNMENT_19  \
City                                                                    
American Fork          3000000.0           3404294.0          15000.0   
Beaver City             400000.0             65884.0              0.0   
Blanding                800000.0             81626.0         800000.0   
Bountiful              4750000.0           2302467.0          15000.0   
Brigham                2000000.0            345233.0           1000.0   

               S_RENTAL_LEASING_19  R_BUILDING_SUPPLY_19  R_CLOTHING_19  \
City                                                                      
American Fork            3494619.0            15701968.0      6160442.0   
Beaver City                87252.0              400000.0        52603.0   
Blanding                  102562.0             1264714.0        47176.0   
Bountiful                1979269.0             1210782.0      2492409.0   
Brigham                  1121260.0             1767578.0       446126.0   

               R_ELECTRONICS_19  R_GROCERY_BEV_19  R_FURNITURE_19  \
City                                                                
American Fork         6990447.0         3872346.0       3623720.0   
Beaver City             21392.0         1500000.0          3000.0   
Blanding                10691.0         1750000.0          8000.0   
Bountiful             3692337.0        19843788.0       1608221.0   
Brigham               1163721.0         9500000.0        493607.0   

               R_GAS_STATIONS_19  R_GE

### Calculate the differences 
*2020 - 2019 = Diff*

In [8]:
# merge the 2019-2020 difference with the sdf
sectors_q1_diff = pd.DataFrame(sectors_q1_2020.values - sectors_q1_2019.values)

# add back the column names
new_names = ['L_ACCOMMODATION_D20', 'S_ADMIN_SUPPORT_D20','A_AG_WILDLIFE_D20','L_CULTURAL_REC_D20','C_CONSTRUCTION_D20',
'E_EDUCATION_D20','S_FINANCIAL_D20','L_RSTRNT_BAR_D20','H_HEALTH_CARE_D20','S_IT_DATA_D20','S_CORPORATE_MGMT_D20',
'M_MANUFACTURING_D20','E_EXTRACTION_D20','O_OTHER_D20','X_ADJUSTMENTS_D20','R_AUTO_PRIVATE_D20','S_PROF_TECH_SRV_D20',
'G_GOVERNMENT_D20','S_RENTAL_LEASING_D20','R_BUILDING_SUPPLY_D20','R_CLOTHING_D20','R_ELECTRONICS_D20','R_GROCERY_BEV_D20',
'R_FURNITURE_D20','R_GAS_STATIONS_D20','R_GENERAL_RETAIL_D20','R_HEALTH_RETAIL_D20','R_OTHER_RETAIL_D20','R_AUTO_RETAIL_D20',
'R_NONSTORE_RETAIL_D20','R_SPORT_HOBBY_D20','L_SPECIAL_EVENT_D20','W_DISTRIBUTION_D20','X_UNKNOWN_D20','U_UTILITIES_D20',
'W_WHLSALE_DURABLE_D20','S_WHLSLE_ETRADE_D20','W_WHLSLE_NDURABLE_D20', 'TOTAL_D20']

sectors_q1_diff.columns = new_names

# add back cities as index
sectors_q1_diff['City'] = sectors_q1_2020.index
sectors_q1_diff = sectors_q1_diff.set_index('City')

#check table
sectors_q1_diff.head(10)

L_ACCOMMODATION_D20  S_ADMIN_SUPPORT_D20  \
City                                                           
American Fork                 -138654.0             -35478.0   
Beaver City                     25895.0              -1325.0   
Blanding                      -129546.0               1719.0   
Bountiful                           0.0            -374373.0   
Brigham                       -391584.0               5169.0   
Cedar City                    -202581.0             289464.0   
Centerville                    -10000.0              43805.0   
Clearfield                     -23250.0               1737.0   
Cottonwood Heights             345039.0             464263.0   
Delta                           50000.0              -1709.0   

                    A_AG_WILDLIFE_D20  L_CULTURAL_REC_D20  C_CONSTRUCTION_D20  \
City                                                                            
American Fork                 50000.0             16648.0           2068027.0   
Beaver City                   15000.0              1000.0            -60000.0   
Blanding                          0.0                 0.0            125597.0   
Bountiful                     14000.0           -160211.0             54379.0   
Brigham                        2000.0                 0.0          -1586913.0   
Cedar City                     1000.0            -26273.0           -558507.0   
Centerville                       0.0             -9000.0             65253.0   
Clearfield                        0.0           1200000.0            450892.0   
Cottonwood Heights             2000.0          -1100000.0             94155.0   
Delta                        100000.0             25000.0           1864078.0   

                    E_EDUCATION_D20  S_FINANCIAL_D20  L_RSTRNT_BAR_D20  \
City                                                                     
American Fork              -11549.0         576097.0          215062.0   
Beaver City                     0.0          56710.0          -57544.0   
Blanding                   115000.0            997.0               0.0   
Bountiful                   24741.0          27960.0         -191766.0   
Brigham                       -17.0         209991.0         -155792.0   
Cedar City                   3728.0        -116105.0          149316.0   
Centerville                  3789.0        -347307.0          394203.0   
Clearfield                  -2387.0         -25057.0         -123919.0   
Cottonwood Heights            269.0         -44971.0        -2858661.0   
Delta                           0.0          34182.0          123759.0   

                    H_HEALTH_CARE_D20  S_IT_DATA_D20  S_CORPORATE_MGMT_D20  \
City                                                                         
American Fork                 -7296.0      -514647.0               -6000.0   
Beaver City                   15000.0      -262738.0                4000.0   
Blanding                      10000.0      -393960.0                   0.0   
Bountiful                    177429.0      1338191.0                   0.0   
Brigham                       -1993.0      -145235.0              -12000.0   
Cedar City                     5729.0       785340.0              -10000.0   
Centerville                  -41064.0      -392471.0              -54000.0   
Clearfield                     -611.0       377238.0                   0.0   
Cottonwood Heights           -50597.0       830941.0                   0.0   
Delta                        -10000.0      -189411.0              -20000.0   

                    M_MANUFACTURING_D20  E_EXTRACTION_D20  O_OTHER_D20  \
City                                                                     
American Fork                 2146185.0          -10000.0    -361365.0   
Beaver City                    474673.0            3000.0      13633.0   
Blanding                       111417.0               0.0      12096.0   
Bountiful                     -245827.0               0.0     141772.0   
Brigham 

In [9]:
# Check join output shape
print(sectors_q1_2019.shape)
print(sectors_q1_2020.shape)
print(sectors_q1_diff.shape)

(66, 39)
(66, 39)
(66, 39)


### Load municipalities and townships shapefile and join to formatted sales tax data

In [10]:
# load cities shapefile into pandas spatial dataframe
cities_shp = '.\\Inputs\\Cities.shp'
cities_sdf = pd.DataFrame.spatial.from_featureclass(cities_shp)

# Rename columns
cities_sdf.columns = ['FID', 'NAME', 'SOURCE', 'SALESTAXID', 'POPLASTCEN', 'POPLASTEST',
       'AREA_SQMI', 'SHAPE']

In [11]:
# merge the 2020 sdf with the pivoted data
cities_sdf2 = cities_sdf.merge(sectors_q1_2019, left_on= "NAME", right_on="City", how="inner")
cities_sdf2 = cities_sdf2.merge(sectors_q1_2020, left_on= "NAME", right_on="City", how="inner")
cities_sdf2 = cities_sdf2.merge(sectors_q1_diff, left_on= "NAME", right_on="City", how="inner")
cities_sdf2.tail(10)


FID              NAME  SOURCE SALESTAXID  POPLASTCEN  POPLASTEST  \
55  222          Syracuse  Cities        049       25113       30400   
56  224      Taylorsville  Cities        142       59989       60192   
57  225       Tooele City  Cities        048       32629       35251   
58  228         Tremonton  Cities        113        7828        8882   
59  232            Vernal  Cities        024        9460       10370   
60  238   Washington City  Cities        027       19640       27686   
61  243    West Bountiful  Cities        061        5420        5731   
62  245       West Jordan  Cities        155      105891      116046   
63  247  West Valley City  Cities        167      133796      136401   
64  252       Woods Cross  Cities        057       10086       11328   

    AREA_SQMI                                              SHAPE  \
55  10.193900  {'rings': [[[411406.60675760085, 4549994.30436...   
56  10.839700  {'rings': [[[422191.826826254, 4504303.3642982...   
57  24.087099  {'rings': [[[392501.67668449413, 4486111.93423...   
58   7.955100  {'rings': [[[399303.9767575987, 4621607.124490...   
59   4.632710  {'rings': [[[627188.8071560834, 4479658.624106...   
60  36.165600  {'rings': [[[278123.416362615, 4102773.6636791...   
61   3.299440  {'rings': [[[421904.10680578306, 4528452.80425...   
62  32.306999  {'rings': [[[415038.796736156, 4499131.1642154...   
63  35.799702  {'rings': [[[418708.9467166035, 4508796.864202...   
64   3.835550  {'rings': [[[424779.70679974474, 4524289.10421...   

    L_ACCOMMODATION_19  S_ADMIN_SUPPORT_19  A_AG_WILDLIFE_19  \
55             30000.0             69731.0            1000.0   
56            300000.0            493259.0            8000.0   
57           1000000.0            175221.0            3000.0   
58            600000.0           1068298.0            2000.0   
59           1361068.0            148452.0            1000.0   
60           2374327.0             79219.0           50000.0   
61            500000.0             11008.0            4000.0   
62           1750000.0            992792.0          350000.0   
63           7060302.0           2906058.0           45000.0   
64           1250000.0            193888.0            1000.0   

    L_CULTURAL_REC_19  C_CONSTRUCTION_19  E_EDUCATION_19  S_FINANCIAL_19  \
55           800000.0           170692.0         27490.0        374231.0   
56          1440305.0           879035.0         79501.0       1971955.0   
57          1250000.0           399845.0         26493.0        886555.0   
58           500000.0           575829.0          7000.0        157020.0   
59           250000.0           163180.0         30688.0        244789.0   
60          1000000.0           876632.0          6158.0        415500.0   
61            70000.0            72248.0          6000.0        118827.0   
62          2405305.0          9010884.0        124595.0       2366371.0   
63          3908740.0         13445854.0        352981.0       4272117.0   
64                0.0          1229952.0         70000.0        245977.0   

    L_RSTRNT_BAR_19  H_HEALTH_CARE_19  S_IT_DATA_19  S_CORPORATE_MGMT_19  \
55        3978564.0           25000.0     2779301.0               7000.0   
56       25781931.0          158751.0     9067000.0              25000.0   
57       11376809.0          696455.0     4736013.0              25000.0   
58        2753334.0            2000.0     1101102.0               7000.0   
59        9176261.0          242065.0     4545387.0              10000.0   
60        8701099.0           20000.0     2789625.0              20000.0   
61        3874568.0            5000.0     1551877.0               3000.0   
62       34384314.0         1165873.0    20867435.0             150000.0   
63       54226479.0         1412358.0    25711264.0              30000.0   
64        4459633.0           20000.0     2133312.0               3000.0   

    M_MANUFACTURING_19  E_EXTRACTION_19  O_OTHER_19  X_ADJUSTMENTS_19  \
55        

In [12]:
# Confirm join count, Salt Lake County (Unincorporated) should be the only one not joined
l1 = list(q1_2020['City'].value_counts().index)
l2 = list(cities_sdf['NAME'].value_counts().index)
list(set(l1) - set(l2))

['Salt Lake County (Unincorporated)']

In [13]:
# Create file gdb and export sdf to feature class
outputs = '.\\Outputs'

gdb = os.path.join(outputs, "taxable_sales.gdb")
if not arcpy.Exists(gdb):
    arcpy.CreateFileGDB_management(outputs, "taxable_sales.gdb")

cities_sdf2.spatial.to_featureclass(location=os.path.join(outputs, "taxable_sales.gdb","taxable_sales_utah_q1"))

'E:\\Projects\\UT_Sales_Tax_Analysis\\Outputs\\taxable_sales.gdb\\taxable_sales_utah_q1'

### Analysis

In [14]:
# Get year specific column names
standard_cols = ['fid','name','source','salestaxid','poplastcen','poplastcen','area_sqmi', 'SHAPE']
names_19 = standard_cols + [col for col in cities_sdf2.columns if '_19' in col]
names_20 = standard_cols + [col for col in cities_sdf2.columns if '_20' in col]
names_D20 = standard_cols + [col for col in cities_sdf2.columns if '_d20' in col]

# subset columns by time
cities_q1_19 = cities_sdf2[names_19]
cities_q1_20 = cities_sdf2[names_20]
cities_q1_d20 = cities_sdf2[names_D20]


In [15]:
# 10 cities with smallest difference of revenue
cities_q1_d20[['name', 'poplastcen', 'area_sqmi', 'total_d20']].sort_values('total_d20', ascending=False).head(10)

name  poplastcen  poplastcen   area_sqmi   total_d20
45    Salt Lake City      190749      190749  110.734001  95830211.0
54         St George       76548       76548   77.209900  61967355.0
50   South Salt Lake       24282       24282    6.936550  51705330.0
63  West Valley City      133796      133796   35.799702  40029181.0
21              Lehi       49315       49315   28.898199  36237576.0
48      South Jordan       51057       51057   22.263201  35667478.0
62       West Jordan      105891      105891   32.306999  33691702.0
51      Spanish Fork       36232       36232   16.249701  30567441.0
18         Kaysville       28050       28050   10.645300  27736795.0
26         Millcreek       62139       62139   12.916500  26248619.0

In [16]:
# 10 cities with largest difference of revenue
cities_q1_d20[['name', 'poplastcen', 'area_sqmi', 'total_d20']].sort_values('total_d20', ascending=True).head(10)

name  poplastcen  poplastcen  area_sqmi   total_d20
35        Park City        7709        7709   20.34170 -35833502.0
59           Vernal        9460        9460    4.63271  -9388974.0
30           Murray       47311       47311   12.29370  -9197282.0
27             Moab        5124        5124    4.80246  -6035165.0
43        Roosevelt        6260        6260    6.75707  -3787091.0
52       Springdale         569         569    4.59488  -2544861.0
12       Farmington       18752       18752   10.03510  -2069172.0
46            Sandy       88693       88693   24.13690  -1503651.0
2          Blanding        3514        3514   13.20550  -1007728.0
32  North Salt Lake       16930       16930    8.51215   -649369.0

#### Some industries of interest:
- RETAIL-GASOLINE STATIONS (447)
- RETAIL-FOOD & BEVERAGE STORES (445)
- REAL ESTATE, RENTAL & LEASING (53)
- ARTS, ENTERTAINMENT AND RECREATION (71)
- ACCOMMODATION (721)